docs/guide/component-testing.md

```diff
@@ -0,0 +1,608 @@
+
+  // Test partial validation
+  await nameInput.fill('John Doe')
+  await submitButton.click()
+
+  await expect.element(getByText('Name is required')).not.toBeInTheDocument()
+  await expect.element(getByText('Email is required')).toBeInTheDocument()
+
+  // Test email format validation
+  await emailInput.fill('invalid-email')
+  await submitButton.click()
+
+  await expect.element(getByText('Please enter a valid email')).toBeInTheDocument()
+
+  // Test successful submission
+  await emailInput.clear()
+  await emailInput.fill('john@example.com')
+  await messageInput.fill('Hello, this is a test message.')
+  await submitButton.click()
+
+  expect(mockSubmit).toHaveBeenCalledWith({
+    name: 'John Doe',
+    email: 'john@example.com',
+    message: 'Hello, this is a test message.'
+  })
+})
+```
+
+### Testing Error Boundaries
+
+```tsx
+// Test how components handle and recover from errors
+function ThrowError({ shouldThrow }: { shouldThrow: boolean }) {
+  if (shouldThrow) {
+    throw new Error('Component error!')
+  }
+  return <div>Component working fine</div>
+}
+
+test('ErrorBoundary catches and displays errors gracefully', async () => {
+  const { getByText, rerender } = render(
+    <ErrorBoundary fallback={<div>Something went wrong</div>}>
+      <ThrowError shouldThrow={false} />
+    </ErrorBoundary>
+  )
+
+  // Initially working
+  await expect.element(getByText('Component working fine')).toBeInTheDocument()
+
+  // Trigger error
+  rerender(
+    <ErrorBoundary fallback={<div>Something went wrong</div>}>
+      <ThrowError shouldThrow={true} />
+    </ErrorBoundary>
+  )
+
+  // Error boundary should catch it
+  await expect.element(getByText('Something went wrong')).toBeInTheDocument()
+})
+```
+
+### Testing Accessibility
+
+```tsx
+test('Modal component is accessible', async () => {
+  const { getByRole, getByLabelText } = render(
+    <Modal isOpen={true} title="Settings">
+      <SettingsForm />
+    </Modal>
+  )
+
+  // Test focus management - modal should receive focus when opened
+  // This is crucial for screen reader users to know a modal opened
+  const modal = getByRole('dialog')
+  expect(document.activeElement).toBe(modal)
+
+  // Test ARIA attributes - these provide semantic information to screen readers
+  expect(modal).toHaveAttribute('aria-labelledby') // Links to title element
+  expect(modal).toHaveAttribute('aria-modal', 'true') // Indicates modal behavior
+
+  // Test keyboard navigation - Escape key should close modal
+  // This is required by ARIA authoring practices
+  await userEvent.keyboard('{Escape}')
+  await waitFor(() => {
+    expect(modal).not.toBeInTheDocument()
+  })
+
+  // Test focus trap - tab navigation should cycle within modal
+  // This prevents users from tabbing to content behind the modal
+  const firstInput = getByLabelText(/username/i)
+  const lastButton = getByRole('button', { name: /save/i })
+
+  firstInput.focus()
```

Comment:
Vitest doesn't have a `focus` method - I already mentioned it before: https://github.com/vitest-dev/vitest/pull/8409#discussion_r2279852202